homeassistant/components/ezviz/camera.py

```diff
@@ -0,0 +1,236 @@
+                conf_cameras,
+            )
+
+        camera["username"] = camera_username
+        camera["password"] = camera_password
+        camera["rtsp_stream"] = camera_rtsp_stream
+
+        camera["ezviz_camera"] = EzvizCamera(ezviz_client, camera_serial)
+
+        camera_entities.append(HassEzvizCamera(**camera))
+
+    add_entities(camera_entities)
+
+
+class HassEzvizCamera(Camera):
+    """An implementation of a Foscam IP camera."""
+
+    def __init__(self, **data):
+        """Initialize an Ezviz camera."""
+        super().__init__()
+
+        self._username = data["username"]
+        self._password = data["password"]
+        self._rtsp_stream = data["rtsp_stream"]
+
+        self._ezviz_camera = data["ezviz_camera"]
+        self._serial = data["serial"]
+        self._name = data["name"]
+        self._status = data["status"]
+        self._privacy = data["privacy"]
+        self._audio = data["audio"]
+        self._ir_led = data["ir_led"]
+        self._state_led = data["state_led"]
+        self._follow_move = data["follow_move"]
+        self._alarm_notify = data["alarm_notify"]
+        self._alarm_sound_mod = data["alarm_sound_mod"]
+        self._encrypted = data["encrypted"]
+        self._local_ip = data["local_ip"]
+        self._detection_sensibility = data["detection_sensibility"]
+        self._device_sub_category = data["device_sub_category"]
+        self._local_rtsp_port = data["local_rtsp_port"]
+
+        self._ffmpeg = None
+
+    def update(self):
+        """Update the camera states."""
+
+        data = self._ezviz_camera.status()
+
+        self._name = data["name"]
+        self._status = data["status"]
+        self._privacy = data["privacy"]
+        self._audio = data["audio"]
+        self._ir_led = data["ir_led"]
+        self._state_led = data["state_led"]
+        self._follow_move = data["follow_move"]
+        self._alarm_notify = data["alarm_notify"]
+        self._alarm_sound_mod = data["alarm_sound_mod"]
+        self._encrypted = data["encrypted"]
+        self._local_ip = data["local_ip"]
+        self._detection_sensibility = data["detection_sensibility"]
+        self._device_sub_category = data["device_sub_category"]
+        self._local_rtsp_port = data["local_rtsp_port"]
+
+    async def async_added_to_hass(self):
+        """Subscribe to ffmpeg and add camera to list."""
+        self._ffmpeg = self.hass.data[DATA_FFMPEG]
+        entities = self.hass.data.setdefault(EZVIZ_DATA, {}).setdefault(ENTITIES, [])
+        entities.append(self)
+
+    @property
+    def should_poll(self) -> bool:
+        """Return True if entity has to be polled for state.
+
+        False if entity pushes its state to HA.
+        """
+        return True
+
+    @property
+    def device_state_attributes(self):
+        """Return the Ezviz-specific camera state attributes."""
+        return {
+            "serial": self._serial,
+            "name": self._name,
+            "status": self._status,
```

Comment:
What does status return?

Reply:
a boolean. Returning if the camera is working or not (plugged in and connected)

Reply:
Since we use it for `available` we don't need it here then.